addons/website/static/src/interactions/header/base_header_special.js

```diff
@@ -0,0 +1,148 @@
+        // 'this.hideEl' element is no longer overflow hidden. Without
+        // this, the dropdown would be invisible.
+        if (this.cssAffixed) {
+            ev.preventDefault();
+            document.scrollingElement.scrollTo({ top: 0, behavior: "smooth" });
+            this.dropdownClickedEl = ev.currentTarget;
+        }
+    }
+
+    onSearchbarInput() {
+        // Prevents the dropdown with search results from being hidden when the
+        // header is fixed.
+        // The scroll animation is instantaneous because the dropdown could open
+        // before reaching the top of the page, which would result in an
+        // incorrect calculated height of the header.
+        if (this.cssAffixed) {
+            document.scrollingElement.scroll({ top: 0 });
+        }
+    }
+
+    onScroll() {
+        super.onScroll();
+
+        const scroll = document.scrollingElement.scrollTop;
+
+        this.atTop = (scroll <= this.topGap);
+        this.isScrolled = (scroll > this.topGap);
+
+        // Need to be 'unfixed' when the window is not scrolled so that the
+        // transparent menu option still works.
+        if (scroll > this.topGap) {
+            if (!this.cssAffixed) {
+                this.transformShow();
+                void this.el.offsetWidth; // Force a paint refresh
+                this.toggleCSSAffixed(true);
+            }
+        } else {
+            this.transformShow();
+            void this.el.offsetWidth; // Force a paint refresh
+            this.toggleCSSAffixed(false);
+        }
+
+        if (this.hideEl) {
+            let elHeight = 0;
+            if (this.cssAffixed && this.searchbarEl?.matches(".show")) {
+                // Close the dropdown of the search bar if it's open when
+                // scrolling. Otherwise, the calculated height of the
+                // 'hideEl' element will be incorrect because it will
+                // include the dropdown height.
+                this.searchbarEl.querySelector("input").blur();
+                elHeight = this.hideEl.offsetHeight;
+            } else {
+                elHeight = this.hideEl.scrollHeight;
+            }
+            const scrollDelta = window.matchMedia(`(prefers-reduced-motion: reduce)`).matches
+                ? scroll
+                : Math.floor(scroll / 4);
+            elHeight = Math.max(0, elHeight - scrollDelta);
+            if (elHeight === 0) {
+                this.hideEl.classList.add("hidden");
+                this.hideEl.removeAttribute("style");
+            } else {
+                // When the page hasn't been scrolled yet, we don't set overflow
+                // to hidden. Without this, the dropdowns would be invisible.
+                // (e.g., "user menu" dropdown).
+                this.hideEl.classList.remove("hidden");
+                this.hideEl.style.overflow = this.cssAffixed ? "hidden" : "";
+                this.hideEl.style.height = this.cssAffixed ? `${elHeight}px` : "";
```

Comment:
Don't we want to do this using `t-att-class` and `t-att-style` ?

Reply:
Because this part is complicated and will some computations, it might be hard to understand the behavior with the use of `t-att`elements.